homeassistant/components/frontier_silicon/media_player.py

```diff
@@ -158,12 +165,19 @@ async def async_update(self):
             }
             self._attr_source_list = list(self.__modes_by_label)
 
-        if not self._attr_sound_mode_list:
-            self.__sound_modes_by_label = {
-                sound_mode.label: sound_mode.key
-                for sound_mode in await afsapi.get_equalisers()
-            }
-            self._attr_sound_mode_list = list(self.__sound_modes_by_label)
+        if not self._attr_sound_mode_list and self._supports_sound_mode:
+            try:
+                self.__sound_modes_by_label = {
+                    sound_mode.label: sound_mode.key
+                    for sound_mode in await afsapi.get_equalisers()
+                }
+                self._attr_sound_mode_list = list(self.__sound_modes_by_label)
```

Comment:
Can you move this out of try... except?
Use try...except...else... format.
Maybe even use an intermediate equalizers variable to move more out of try...except.

Reply:
I've reduced the length of both try-blocks to only contain the statements that can effectively throw that error.